From 2419bd453bd530a6d104cf6bb63d3f6e07c061d8 Mon Sep 17 00:00:00 2001
From: Ioan-Adrian Ratiu <adrian.ratiu@ni.com>
Date: Tue, 29 Mar 2016 17:26:19 +0300
Subject: [PATCH 1/2] Define SMBIOS3 entry point structures for EFI

This adds the GUID, and includes it in lsefisystab output.

Backported from patch series http://marc.info/?l=grub-devel&m=145749821829048
Rebased on top of grub-legacy

Upstream-Status: Submitted

Signed-off-by: Ioan-Adrian Ratiu <adrian.ratiu@ni.com>
---
 grub-core/commands/efi/lsefisystab.c | 1 +
 include/grub/efi/api.h               | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/grub-core/commands/efi/lsefisystab.c b/grub-core/commands/efi/lsefisystab.c
index b235925..33d4699 100644
--- a/grub-core/commands/efi/lsefisystab.c
+++ b/grub-core/commands/efi/lsefisystab.c
@@ -39,6 +39,7 @@ static const struct guid_mapping guid_mappings[] =
     { GRUB_EFI_ACPI_TABLE_GUID, "ACPI-1.0"},
     { GRUB_EFI_SAL_TABLE_GUID, "SAL"},
     { GRUB_EFI_SMBIOS_TABLE_GUID, "SMBIOS"},
+    { GRUB_EFI_SMBIOS3_TABLE_GUID, "SMBIOS3"},
     { GRUB_EFI_MPS_TABLE_GUID, "MPS"},
     { GRUB_EFI_HCDP_TABLE_GUID, "HCDP"}
   };
diff --git a/include/grub/efi/api.h b/include/grub/efi/api.h
index 26127de..79071a5 100644
--- a/include/grub/efi/api.h
+++ b/include/grub/efi/api.h
@@ -124,6 +124,11 @@
     { 0x9a, 0x16, 0x0, 0x90, 0x27, 0x3f, 0xc1, 0x4d } \
   }
 
+#define GRUB_EFI_SMBIOS3_TABLE_GUID	\
+  { 0xf2fd1544, 0x9794, 0x4a2c, \
+    { 0x99, 0x2e, 0xe5, 0xbb, 0xcf, 0x20, 0xe3, 0x94 } \
+  }
+
 #define GRUB_EFI_SAL_TABLE_GUID \
   { 0xeb9d2d32, 0x2d88, 0x11d3, \
       { 0x9a, 0x16, 0x0, 0x90, 0x27, 0x3f, 0xc1, 0x4d } \
-- 
2.9.0

